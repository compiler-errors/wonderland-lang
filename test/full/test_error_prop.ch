// full pass

fn try_sum(a: Result<Int, Char>, b: Result<Int, Char>) -> Result<Int, Char> {
    Result!Ok(a? + b?)
}

fn print_try_sum(a: Result<Int, Char>, b: Result<Int, Char>) {
    println("\(a) + \(b) = \(try_sum(a, b))").
}

fn main() -> Int {
    print_try_sum(Result!Error('a'), Result!Error('b')).
    print_try_sum(Result!Ok(1),      Result!Error('b')).
    print_try_sum(Result!Error('a'), Result!Ok(1)).
    print_try_sum(Result!Ok(1),      Result!Ok(2)).

    0
}