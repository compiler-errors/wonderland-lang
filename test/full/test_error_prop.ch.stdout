Error(a) + Error(b) = Error(a)
Ok(1) + Error(b) = Error(b)
Error(a) + Ok(1) = Error(a)
Ok(1) + Ok(2) = Ok(3)